### DISCUSSION TITLE
Replace veux $store with $pinia

### DISCUSSION THREAD

**Discussion by rendomnet** [vuejs/pinia#1958] [Help and Questions] [answered]
When using vuex we have $store object that is accessible in every component
How can I have the same with pinia?
Basicaly I want to have $store pinia object available in every component same as vuex.

**Comment by posva** [comment, marked as answer]
You already have `$pinia` available, which gives you access to all instantiated stores' state:

```js
this.$pinia.state.value
```

If you want access to the stores, you will need to collect them in a plugin and add them to global properties:

```ts
const stores = {}
pinia.use(({ store }) => {
  stores[store.$id] = store
})
app.config.globalProperties.$stores = stores
```

But I recommend you to avoid this because it cannot be made type safe and stays in an _"vuex way of thinking"_ instead of embracing the new defaults that enable better code splitting and dynamic registration by default
